tests/providers/test_automotive.py

```diff
@@ -146,3 +146,28 @@ def test_fr_FR_plate_format(self):
         plate = self.fake.license_plate()
         assert isinstance(plate, str)
         assert self.pattern.match(plate)
+
+
+class TestEsES(unittest.TestCase):
+
+    def setUp(self):
+        self.fake = Faker('es_ES')
+        Faker.seed(0)
+        self.new_format_pattern = re.compile(r'\d{4}\s[A-Z]{3}')
+        self.old_format_pattern = re.compile(r'[A-Z]{1,2}\s\d{4}\s[A-Z]{2}')
+
+    def test_es_ES_plate_new_format(self):
+        plate = self.fake.license_plate_unified()
+        assert isinstance(plate, str)
+        assert self.new_format_pattern.match(plate)
+
+    def test_es_ES_plate_old_format(self):
+        plate = self.fake.license_plate_by_province()
```

Comment:
Can you add a test that calls `license_plate_by_province()` with a province?

Reply:
It's done.